CoreFeatures/Database-postgresql-plugins/PluginPostgresDBTasks/src/main/java/info/smart_tools/smartactors/database_postgresql_plugins/postgres_db_tasks_plugin/PostgresDBTasksPlugin.java

```diff
@@ -503,6 +506,88 @@ public IObject getCriteria() throws ReadValueException {
+                                        throw new ReadValueException("Failed to get search criteria from the message", e);
+                                    }
+                                }
+
+                                @Override
+                                public IObject getPercentileCriteria() throws ReadValueException {
+                                    try {
+                                        return percentileCriteriaField.in(message);
+                                    } catch (InvalidArgumentException e) {
+                                        throw new ReadValueException("Failed to get percentile search criteria from the message", e);
+                                    }
+                                }
+
+                                @Override
+                                public IAction<Number[]> getCallback() throws ReadValueException {
+                                    try {
+                                        return (IAction<Number[]>) callbackField.in(message);
+                                    } catch (InvalidArgumentException e) {
+                                        throw new ReadValueException("Failed to get callback from the message", e);
+                                    }
+                                }
+                            };
+                        }
+                )
+        );
+        IOC.register(
+                Keys.getKeyByName("db.collection.percentileSearch"),
+                //TODO:: use smth like ResolveByNameStrategy, but this caching strategy should call prepare always
+                new ApplyFunctionToArgumentsStrategy(
+                        (args) -> {
+                            try {
+                                IStorageConnection connection = (IStorageConnection) args[0];
+                                CollectionName collectionName = CollectionName.fromString(String.valueOf(args[1]));
+                                IObject criteria = (IObject) args[2];
+                                IObject percentileCriteria = (IObject) args[3];
+                                IAction<IObject[]> callback = (IAction<IObject[]>) args[4];
+                                IDatabaseTask task = new PostgresPercentileSearchTask(connection);
+
+                                IObject query = IOC.resolve(Keys.getKeyByName(IObject.class.getCanonicalName()));
+
+                                collectionNameField.out(query, collectionName);
+                                criteriaField.out(query, criteria);
+                                percentileCriteriaField.out(query, percentileCriteria);
+                                callbackField.out(query, callback);
+
+                                task.prepare(query);
+                                return task;
+                            } catch (Exception e) {
+                                throw new RuntimeException("Can't resolve search db task.", e);
```

Comment:
Shouldn't a checked exception be thrown here?

Reply:
You're right, there supposed to be checked exception instead of unchecked. Thanks for noticing that.
Replaced it with `FunctionExectionException`